### PR TITLE
`General`: Sort courses alphabetically by title

### LINKED ISSUE #4239: Sort the courses of a semester by name
### Is your feature request related to a problem?

_No response_

### Describe the solution you'd like

I would suggest that the lectures of a semester should be ordered by their names. Otherwise its hard to find them (iff you have many)

Course Overview:
Course_Overview

Course Management:
Course_Mgmt

Course Registration:
Course_Register

I've tagged it as small since I assume that you simply have to add `.sorted()` somewhere in the UI.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ole-ve**
#### General
- [ ] This is a small issue that I tested locally and was confirmed by another developer on a test server
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context
This pull request closes #4239. Courses on course overview and management dashboard are sorted alphabetically ascending by their title.

### Steps for Testing

Prerequisites:
- 1 Instructor

1. Log in to Artemis
2. Navigate to Course Management
3. Verify that courses within the same semester are sorted correctly.
4. Navigate to Course Overview
5. Click '# Register for a course'
6. Verify that courses are sorted correctly.

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots
image
image
image

### CODE REVIEW

File: src/main/webapp/app/overview/course-registration-selector/course-registration-selector.component.ts [hunk 1]
```diff
@@ -45,9 +45,11 @@ export class CourseRegistrationSelectorComponent implements OnInit {
         return new Promise<void>((resolve, reject) => {
             this.courseService.findAllToRegister().subscribe(
                 (registerRes) => {
-                    this.coursesToSelect = registerRes.body!.filter((course) => {
-                        return !this.courses.find((el) => el.id === course.id);
-                    });
+                    this.coursesToSelect = registerRes
+                        .body!.filter((course) => {
```
Comment (Hialus) [inline src/main/webapp/app/overview/course-registration-selector/course-registration-selector.component.ts:49]: I think the `.body!` part should either be on its own line or on the same line as `registerRes`

**Review by Hialus — CHANGES_REQUESTED** [review]
Tested on TS1 and works as expected.
Code changes look fine apart from one small issue.
Well done 👍🏻

File: src/main/webapp/app/overview/course-registration-selector/course-registration-selector.component.ts (hunk 1 above)
Comment (ole-ve) [reply src/main/webapp/app/overview/course-registration-selector/course-registration-selector.component.ts:49]: ESLint actually does not think so, changing to proposed solutions leads to code style errors.

**Review by Olimbia23 — DISMISSED** [review]
Nice work @ole-ve.
Code changes look good.
Approve 👍

File: src/main/webapp/app/overview/course-registration-selector/course-registration-selector.component.ts (hunk 1 above)
Comment (Hialus) [reply src/main/webapp/app/overview/course-registration-selector/course-registration-selector.component.ts:49]: Okay. It's just not the code style I'm used to.

**Review by Hialus — DISMISSED** [review]
As stated above everything works fine 👍🏻

**Review by DanielKainz — APPROVED** [review]
I tested it on TS1. The courses are sorted correctly :-)

**Review by Hialus — APPROVED** [review]
Retested on TS2.
Still works as expected 👍🏻

**Review by filip-gregurevic — APPROVED** [review]
Looks good to me 👍🏼

**Pull request merged**